src/builder/modules/Generator_Builder.res

```diff
@@ -41,6 +41,24 @@ type rectangleLegacy = {
 
 type rectangle = (int, int, int, int)
 
+type cuboid = {
+  top: rectangle,
+  bottom: rectangle,
+  front: rectangle,
+  right: rectangle,
+  left: rectangle,
+  back: rectangle,
+}
+
+let makeCuboid = ((x, y): position, (w, h, l)): cuboid => {
```

Comment:
Nice! Small suggestion - both width and height are also "lengths", so could I suggest we use width, height, and _depth_.

Also, I did a quick diagram - can you check if this is correct and add it as a comment above this function?

```
W = Width
H = Height
D = Depth

(x,y)            
  .         +----W----+    
            |         |
            D   Top   D 
            |         |
  +----D----+----W----+----D----+----W----+
  |         |         |         |         |
  H  Right  H  Front  H  Left   H  Back   H
  |         |         |         |         |
  +----D----+----W----+----D----+----W----+
            |         |
            D Bottom  D
            |         |
            +----W----+    
```

A few thoughts:

I'm thinking about separate "creating" cuboids from "positioning" cuboids.

For example:
```
// Maybe move into a module
module Cuboid = {
 type t = ... // cuboid type goes here
  let make = (width: int, height: int, depth: int): t => { } // Creates a new cubiod, independent of position.
  let translate = (cuboid: t, x: int, y: int): t => { } // Translates the cuboid by the given x and y coordinates.
} 
```

Example usage:

```res
Cuboid.make(3, 12, 4)->Cuboid.translate(40, 16)
```

Not sure, just thinking about it at the moment.

Reply:
> Nice! Small suggestion - both width and height are also "lengths", so could I suggest we use width, height, and depth.

From what I've gathered, "height", "width", "depth", "breadth", and "length" can be used interchangeably, but in the cases for the dimensions of a box, such as a papercraft cuboid, I've seen width and length be used for the horizontal axes, and either height or depth be used for the vertical axis. However, it is minor, and width and length could get easily confused for each other, while "depth" might feel more distinct from the other two. I'll think about this.

> I'm thinking about separate "creating" cuboids from "positioning" cuboids.

I think this is the solution I was looking for earlier- I wanted to have makeCuboid() be just the constructor for the cuboid type, but I couldn't figure out what that would look like while still having cuboid be a record of rectangles. I also wanted to have cuboids be used for the destination in drawCuboid(), but I ended up not doing so since there was not preexisting map for body parts. These might be added to `Minecraft.res` when it is made.